components/salsa-2022/src/runtime/local_state.rs

```diff
@@ -114,28 +112,35 @@ pub struct QueryEdges {
     /// Important:
     ///
     /// * The inputs must be in **execution order** for the red-green algorithm to work.
-    /// * The outputs must be in **sorted order** so that we can easily "diff" them between revisions.
-    input_outputs: Arc<[DependencyIndex]>,
-
-    /// The index that separates inputs from outputs in the `tracked` field.
-    separator: u32,
+    pub input_outputs: Arc<[(EdgeKind, DependencyIndex)]>,
 }
 
 impl QueryEdges {
     /// Returns the (tracked) inputs that were executed in computing this memoized value.
     ///
     /// These will always be in execution order.
-    pub(crate) fn inputs(&self) -> &[DependencyIndex] {
-        &self.input_outputs[0..self.separator as usize]
+    pub(crate) fn inputs(&self) -> Vec<DependencyIndex> {
```

Comment:
What code calls this now?

Reply:
It is only called here.  
https://github.com/salsa-rs/salsa/blob/07848c2818077635658ba94b245d6f5713909452/components/salsa-2022/src/function/accumulated.rs#L66-L76